lib/topological_inventory/ingress_api/messaging_client.rb

```diff
@@ -0,0 +1,38 @@
+module TopologicalInventory
+  module IngressApi
+    def self.with_messaging_client
+      messaging_client ||= raw_messaging_client
+      raise if messaging_client.nil?
+
+      begin
+        yield messaging_client
+      rescue Kafka::Error
+        # If we hit an underlying kafka error then reset the connection
+        messaging_client.close
+        self.messaging_client = nil
+
+        raise
+      rescue Kafka::MessageSizeTooLarge
```

Comment:
I think this needs to be above the superclass rescue block.

Reply:
Or you could make the handling conditional on the error class in a single block .. Your call

Reply:
yeah you're right, fixed